packages/repo-metadata-lint/src/repo-metadata-lint.ts

```diff
@@ -26,8 +26,27 @@ export function handler(app: Probot) {
   // eslint-disable-next-line @typescript-eslint/no-explicit-any
   app.on('schedule.repository' as any, async context => {
     const owner = context.payload.organization.login;
-    // const repo = context.payload.repository.name;
-    logger.info('schedule endpoint', owner);
+    const repo = context.payload.repository.name;
+    const iterator = new fileIterator.FileIterator(
+      owner,
+      repo,
+      context.octokit
+    );
+    const results: Array<ValidationResult> = [];
+    const validate = new Validate(context.octokit);
+    for await (const [path, metadata] of iterator.repoMetadata()) {
```

Comment:
nit: an async iterator certainly works here, but each file validation is an independent event and there's no need to stop iteration until we've validated every file. Instead, we could do a basic loop, collect promises, and `await Promise.all([promises])`. They are also more of a pain to stub for testing.

Reply:
Yeah I agree, okay if I refactor when I write some more tests, just want to test this end to end.